devito/data/data.py

```diff
@@ -179,9 +179,9 @@ def __str__(self):
         return super(Data, self._local).__str__()
 
     @_check_idx
-    def __getitem__(self, glb_idx, comm_type):
+    def __getitem__(self, glb_idx, comm_type, gather=None):
         loc_idx = self._index_glb_to_loc(glb_idx)
-        if comm_type is index_by_index:
+        if comm_type is index_by_index or isinstance(gather, int):
```

Comment:
Should we have a default `gather=True` equivalent to `gather=0`? Or is it overkill